examples/vision/supervised-contrastive-learning.py

```diff
@@ -0,0 +1,204 @@
+"""
+Title: Supervised Contrastive Learning
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2020/11/01
+Last modified: 2020/11/01
+Description: Using supervised contrastive learning for image classification.
+"""
+
+"""
+## Introduction
+
+[Supervised Contrastive Learning](https://arxiv.org/abs/2004.11362)
+(Prannay Khosla et al.) is a training methodology that outperforms cross-entropy
+on supervised learning tasks.
+
+Essentially, training an image classification model with Supervised Contrastive Learning
+is peformed in two phases:
+
+  1. Pre-training an encoder to generate feature vectors for input images such that feature
+    vectors of images in the same class will be more similar compared feature vectors of
+    images in other classes.
+  2. Training a classifier on top of the freezed encoder.
+
+"""
+
+"""
+## Setup
+"""
+
+"""shell
+!pip install tensorflow-addons
+"""
+
+import tensorflow as tf
+import tensorflow_addons as tfa
+import numpy as np
+
+"""
+## Prepare the data
+"""
+
+num_classes = 10
+input_shape = (32, 32, 3)
+
+# the data, split between train and test sets
+(x_train, y_train), (x_test, y_test) = tf.keras.datasets.cifar10.load_data()
+
+y_train, y_test = tf.squeeze(y_train), tf.squeeze(y_test)
```

Comment:
This line is unexplained

Reply:
Explanation added.

Reply:
This isn't really necessary, is it? What happens if you remove the line?

Reply:
Seems like it isn't :-) 
Removed.